### DISCUSSION TITLE
Warnings in testing environment about usage of `onBeforeRouteLeave` without an active route record

### DISCUSSION THREAD

**Discussion by Sensanaty** [vuejs/router#2328] [Help and Questions] [answered]
`router` version `4.2.4`, Vue version `3.3.4`.

I've got a fairly basic set of unit tests testing a route component. The tests themselves seem to be working just fine, but I do get an error on every single test about my usage of `onBeforeRouteLeave` in the component, the specific warning being:

```
[Vue Router warn]: No active route record was found when calling `onBeforeRouteLeave()`. Make sure you call this function inside a component child of <router-view>. Maybe you called it inside of App.vue?
```

I have my tests set up as [described in the docs](https://test-utils.vuejs.org/guide/advanced/vue-router#Using-a-real-router-with-Composition-API), like so:

```ts
describe('MyComponent.vue', () => {
  let router: Router;

  beforeEach(() => {
    router = createRouter({ history: createWebHistory(), routes }); // routes is my main routes file used in the app
  })

  it('should render', async () => {
    await router.push({ name: 'my-route' });
    await router.isReady();

    component = mount(MyComponent, { global: { plugins: [router] } };
    expect(component.exists()).toBe(true);
  })
}
```

Any clues on what I'm doing wrong here? The component itself isn't doing anything crazy with the `onBeforeRouteLeave()`, I just reset some pinia stores, but this functionality is also tested and works fine in the tests, though it also leaves the same warning.

**Comment by posva** [comment, marked as answer]
In short, `MyComponent` should have a `<RouterView />` that should be rendering the component that contains the `onBeforeRouteLeave()`, otherwise it won't register the guard properly
